### PR TITLE
feat: allow users to choose their own table name

### PR DESCRIPTION
**Pull request by niross**
### Description of change

- If the table name is invalid, allow it to be corrected
- If the table already exists in their schema, confirm overwrite
- If the table does not exist, run the DAG

closes: https://trello.com/c/spFdGRVN/1391-csv-upload-choose-table-name-table-already-exists-page-4

### Checklist

* [x] Have tests been added to cover any changes?

### Confirm
Screenshot from 2021-02-12 10-05-11

### Confirm table name
Screenshot from 2021-02-12 10-05-19

### Invalid table name
Screenshot from 2021-02-12 10-28-49

### Table exists
Screenshot from 2021-02-12 10-29-02

### CODE REVIEW

File: dataworkspace/dataworkspace/templates/your_files/create-table-table-exists.html [hunk 1]
```diff
@@ -7,19 +7,18 @@
 {% endblock %}
 
 {% block content %}
+  {{ form.errors }}
```
Comment (samuelhwilliams) [inline dataworkspace/dataworkspace/templates/your_files/create-table-table-exists.html:10]: I think this should still use the error summary partial?

Comment (samuelhwilliams) [reply dataworkspace/dataworkspace/templates/your_files/create-table-table-exists.html:10]: Ah, this page "shouldn't" really ever throw validation errors right? 🤔  So probably fine.

File: dataworkspace/dataworkspace/templates/your_files/create-table-table-exists.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          </strong>
+        </div>
+          {% csrf_token %}
+          {{ form.table_name }}
+          {{ form.path }}
+          {{ form.schema }}
+          {{ form.force_overwrite }}
+        </div>
+        <button type="submit" class="govuk-button">Yes, I understand - Continue</button>
+        <a href="{% url 'your_files:create-table-confirm-name' %}?path={{ request.GET.path }}&table_name={{ form.table_name.value }}" class="govuk-button govuk-button--secondary">
```
Comment (samuelhwilliams) [inline dataworkspace/dataworkspace/templates/your_files/create-table-table-exists.html:37]: Does this also need to send the schema to populate the back form correctly?

File: dataworkspace/dataworkspace/templates/your_files/create-table-table-exists.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      <form method="POST" action="{% url 'your_files:create-table-confirm-name' %}?{{ request.GET.urlencode }}" novalidate>
+        <div class="govuk-form-group{% if form.table_name.errors %} govuk-form-group--error{% endif %}">
+          <h1 class="govuk-label-wrapper">
+            <label class="govuk-label govuk-label--xl">
+              Table already exists
+            </label>
+          </h1>
+          <div class="govuk-hint">
+            A table already exists in your personal database schema with the name "{{ form.table_name.value }}". Do you want to overwrite the existing table?
+          </div>
```
Comment (samuelhwilliams) [inline dataworkspace/dataworkspace/templates/your_files/create-table-table-exists.html:22]: I can't quite put my finger on it, but this HTML feels a bit off to me. I don't think we need the `label` element, particularly as we're not pointing the label at another element.

I wonder if maybe we just need to have `aria-describedby=<some_id>` on the "Continue" button, where <some_id> is the id of the div that describes what happens ("A table already exists ... do you want to overwrite?"). That feels like it provides more context on the action.

**Review by samuelhwilliams — APPROVED** [review]

File: dataworkspace/dataworkspace/templates/your_files/create-table-table-exists.html (hunk 2 above)
Comment (niross) [reply dataworkspace/dataworkspace/templates/your_files/create-table-table-exists.html:37]: Indeed, good catch :+1:

**Pull request merged**